packages/geometry/src/Matrix.ts

```diff
@@ -0,0 +1,27 @@
+export class Matrix {
+	matrix: number[][];
+
+	constructor(matrix: number[][]) {
+		this.matrix = matrix;
+	}
+
+	mmul = (matrix: Matrix): Matrix => {
+		// prettier-ignore
+		this.matrix = this.matrix.map((row, i) =>
+				matrix.asArray()[0].map((_, j) =>
+					row.reduce((acc, _, n) =>
+						acc + this.matrix[i][n] * matrix.asArray()[n][j], 0
+					)
+				)
+			);
+		return this;
+	};
+
+	asArray = (): number[][] => {
```

Comment:
these should be changed to actual class methods otherwise the class holds no meaning and new instances of the methods are created on every instance of the class. But I'm going to just make this change myself, was more an FYI. (this kind of pattern is semi useful in react classes but for different reasons)